### DISCUSSION TITLE
Pug support bug?

### DISCUSSION THREAD

**Discussion by ivanq3w** [windicss/windicss#423] [Q&A]
Doesn`t work: windicss classes styles does not applied
```
<template lang="pug">
h1.text-white.bg-red-500.m-10 Hello world
</template>
```
Works: windicss classes styles applied
```
<template lang="pug">
h1(class="text-white bg-red-500 m-10") Hello world
</template>
```
Why windicss cannot parse pug inline classes?

**Comment by ivanq3w** [comment]
I have a suspicion that pug classes (.text-white.bg-red-500.m-10) works in node version 16+ and doesnt works in node version 14.

**Comment by ivanq3w** [comment]
And second suspicion is windows!
If you dont do this before cloning the project from git.
```
git config --global core.autocrlf false
```
A lot of libs works wrong... eslint, prettier and windicss.

**Reply by ivanq3w** [reply]
FYI https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

**Comment by chamberlainpi** [comment]
Would this even work in Pug's shorthand class list when using WindiCSS's Variant Groups?

Ex: `hover:(bg-gray-400 font-medium) bg-white font-light`

Since "hover" is not a class perse, it doesn't have a `.` as a starting-character (as a CSS class naturally does).

And as soon as you follow your `<HTMLElement>` name with a space ` `, well it's going to interpret that as the text content, right?

Ex: `h1 hover:(.bg-gray-400.font-medium).bg-white.font-light` would result:
`<h1>hover:(.bg-gray-400.font-medium).bg-white.font-light</h1>` 

Correct me if I'm wrong!

So maybe it's best to stick to the slightly longer syntax when using Pug, that is:
`h1(class="hover:(bg-gray-400 font-medium) bg-white font-light") Hello World`

Has this been the general consensus on using Pug with WindiCSS?
